sourcegraph/sourcegraph
# Cody: Re-spin tasks which have conflicts with local edits

This does not *automatically* re-spin tasks, but the respin mechanism is working.

## Test plan

```
pnpm test:unit && pnpm test:integration && pnpm test:e2e
```

Manual test:

1. Create and save a file. Write a bunch of code in it.
2. Ask Cody to do something time consuming in chat, like list US states and their capitals.
3. While chat is busy, Cmd-Shift-P, Fixup (Experimental) and ask Cody to fix something. Note that the fixup task is paused while chat is busy, and gets picked up automatically when chat is not busy.
4. Edit in a blue rectangle to create a conflict with what Cody wants to write. Click Apply. Note that the task re-spins and Cody redoes its work including your changes.
5. Edit "in the zone" but outside blue rectangles so there's no conflict. Apply the change. Note that all the changes are applied, merged.
